### PR TITLE
Remove column api. JB#56729

### PR DESCRIPTION
**Pull request by pvuorela**
From the main commit: 

    The models had a strange dual mode with two-dimensional table having
    different data on different columns, but also one-dimensional list
    type of character with different data avialable with roles.
    
    By the history it was first having column ids, and later got the more
    conventional roles, somewhat randomly new data added to either.
    
    Killed here the column based access. It wasn't available to QML in the
    first place either.

While at it, noticed bunch of other small details to adjust. Separate commits for more info. Including just cosmetics changes too.

@Tomin1 @llewelld

### CODE REVIEW

File: src/eventtreeitem.cpp [hunk 1]
```diff
@@ -49,17 +49,16 @@ void EventTreeItem::prependChild(EventTreeItem *child)
...
 
-void EventTreeItem::moveChild( int fromRow, int toRow )
+void EventTreeItem::moveChild(int fromRow, int toRow)
 {
     // if the given index is out of range or the two index are the same
-    if ( fromRow < 0 || childCount() - 1 < fromRow ||
-         toRow < 0 || childCount() -1 < toRow ||
-         fromRow == toRow )
-    {
+    if (fromRow < 0 || childCount() - 1 < fromRow ||
```
Comment (Tomin1) [inline src/eventtreeitem.cpp]: I think that last `||` belongs to next line. It's the same on the next row.

File: src/eventmodel.cpp [hunk 2]
```diff
@@ -59,27 +59,27 @@ EventModel::~EventModel()
 QHash<int, QByteArray> EventModel::roleNames() const
 {
     QHash<int,QByteArray> roles;
```
Comment (Tomin1) [inline src/eventmodel.cpp:61]: This is missing `ContactsRole`.

File: src/eventmodel.cpp [hunk 3]
```diff
@@ -306,83 +304,77 @@ QVariant EventModel::data(const QModelIndex &index, int role) const
...
     }
 
-    int column = index.column();
-    if (role >= BaseRole) {
-        column = role - BaseRole;
-        role = Qt::DisplayRole;
-    }
-
     QVariant var;
-    switch (column) {
```
Comment (Tomin1) [inline src/eventmodel.cpp:316]: Should this `switch` be combined with the one above? It seems a bit odd to have two when there is no obvious reason to keep both.

File: src/groupmodel.cpp [hunk 4]
```diff
@@ -280,65 +272,55 @@ QVariant GroupModel::data(const QModelIndex &index, int role) const
...
-    }
-    
-    if (role != Qt::DisplayRole) {
-        return QVariant();
-    }
-
     QVariant var;
-    switch (column) {
-        case GroupId:
+    switch (role) {
```
Comment (Tomin1) [inline src/groupmodel.cpp]: Also here it seems a little strange to do first if comparisons to `role` and then have a switch. Maybe these could be both in the same switch.

File: src/groupmodel.cpp [hunk 5]
```diff
@@ -280,65 +272,55 @@ QVariant GroupModel::data(const QModelIndex &index, int role) const
...
             break;
-        case StartTime:
+        case StartTimeRole:
             var = QVariant::fromValue(group->startTime());
             break;
         default:
-            DEBUG() << "Group::data: invalid column id??" << column;
+            DEBUG() << "Group::data: invalid role??" << role;
             break;
     }
```
Comment (Tomin1) [inline src/groupmodel.cpp]: `IsPermanentRole` is not handled.

File: src/contactgroupmodel.cpp [hunk 6]
```diff
@@ -385,74 +378,65 @@ QVariant ContactGroupModel::data(const QModelIndex &index, int role) const
...
         return g->endTime().toLocalTime().date();
     }
 
-    if (role != Qt::DisplayRole)
-        return QVariant();
-
     QVariant var;
-    switch (column) {
-        case ContactIds:
+    switch (role) {
```
Comment (Tomin1) [inline src/contactgroupmodel.cpp]: That `if` + `switch` structure is also here.

**Comment by Tomin1** [comment]
The comment was left out, but I was going to say that these are mostly cosmetic and some of these things were already there before and weren't noticed.

File: src/eventtreeitem.cpp (hunk 1 above)
Comment (pvuorela) [reply src/eventtreeitem.cpp]: Sure. Didn't adjust the end of the line operators that much now, more of the extra spaces and strange braces that were standing out more clearly, but I'll adjust.

File: src/eventmodel.cpp (hunk 3 above)
Comment (pvuorela) [reply src/eventmodel.cpp:316]: Guess I could just merge. Kept the original structure to limit the delta a bit, but it shouldn't be that complex change so I'll just do.

File: src/eventmodel.cpp (hunk 2 above)
Comment (pvuorela) [reply src/eventmodel.cpp:61]: Nicely spotted. Also at least EventRole. I'll readjust these to match the enum declaration order so it's easier to keep track.

File: src/groupmodel.cpp (hunk 5 above)
Comment (pvuorela) [reply src/groupmodel.cpp]: Interesting. Wasn't handled earlier either, so guess that can be just ditched.

**Comment by pvuorela** [comment]
Adjusted the commented details, e.g. merged the role -> data mapping. IsPermanent removed as non-functional.

**Review by Tomin1 — APPROVED** [review]
LGTM. I didn't check the build system changes quite as carefully as the rest but it does build and it looked all-right.

**Pull request merged**